specification/trafficmanager/resource-manager/Microsoft.Network/stable/2018-04-01/trafficmanager.json

```diff
@@ -1098,34 +1107,111 @@
       },
       "description": "Class representing a Traffic Manager HeatMap traffic flow properties."
     },
-    "HeatMapProperties": {
+    "HeatMapPropertyBase": {
+      "type": "object",
+      "discriminator": "heatMapType",
       "properties": {
-        "startTime": {
-          "type": "string",
-          "format": "date-time",
-          "description": "The beginning of the time window for this HeatMap, inclusive."
+        "heatMapType": {
+          "type": "string"
+        }
+      },
+      "required": [
+        "heatMapType"
+      ],
+    },
+    "default":{
```

Comment:
Use a meaingful name, and the `x-ms-discriminator-value` for the real value. Example:
```json
    "DatasetBZip2Compression": {
      "x-ms-discriminator-value": "BZip2",
```
This means "Create class DatasetBZip2Compression, but match value BZip2 on the wire"

Reply:
I see. so here I can use  
"HeatMapPropertyAsn": 
   {"x-ms-discriminator-value": "asn",
 to indicate that when heatMapType=="asn", the HeatMapPropertyAsn will be applied? 
if so, that's very convenient - I was very confused how I can link the two.